packages/ckeditor5-ui/src/button/listitembuttonview.ts

```diff
@@ -0,0 +1,195 @@
+		super( locale, labelView );
+
+		this.set( {
+			hasReservedCheckHolderSpace: false,
+			_shouldRenderCheckHolder: this.isToggleable
+		} );
+
+		const bind = this.bindTemplate;
+
+		this.extendTemplate( {
+			attributes: {
+				class: [
+					'ck-list-item-button',
+					bind.if( 'isToggleable', 'ck-list-item-button_toggleable' )
+				]
+			}
+		} );
+
+		this.bind( '_shouldRenderCheckHolder' ).to(
+			this, 'hasReservedCheckHolderSpace',
+			this, 'isToggleable',
+			( hasReservedCheckHolderSpace, isToggleable ) => hasReservedCheckHolderSpace || isToggleable
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		if ( this._shouldRenderCheckHolder ) {
+			this.children.add( this._checkIconHolderView, 0 );
+		}
+
+		this._watchCheckIconHolderMount();
+	}
+
+	/**
+	 * Renders the check icon if the button is toggleable.
+	 */
+	private _watchCheckIconHolderMount() {
+		this._checkIconHolderView
+			.bind( 'isOn' )
+			.to( this, 'isOn', value => this.isToggleable && value );
+
+		this.on<ObservableChangeEvent<boolean>>(
+			'change:_shouldRenderCheckHolder',
+			( evt, propertyName, shouldRenderCheckHolder ) => {
+				const { children, _checkIconHolderView } = this;
+
+				if ( shouldRenderCheckHolder && !children.has( _checkIconHolderView ) ) {
+					children.add( _checkIconHolderView, 0 );
+				} else if ( !shouldRenderCheckHolder && children.has( _checkIconHolderView ) ) {
+					children.remove( _checkIconHolderView );
+				}
+			}
+		);
+	}
+}
+
+export class CheckIconHolderView extends View {
+	/**
+	 * The view for the check icon of the button list item.
+	 */
+	private readonly _checkIconView: IconView = this._createCheckIconView();
+
+	/**
+	 * Collection of child views.
+	 */
+	public readonly children: ViewCollection<View>;
+
+	/**
+	 * Indicates whether the button is in the "on" state.
+	 */
+	declare public isOn: boolean;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor() {
+		super();
+
+		const bind = this.bindTemplate;
+
+		this.children = this.createCollection();
+
+		this.set( 'isOn', false );
+		this.setTemplate( {
+			tag: 'span',
+			children: this.children,
+			attributes: {
+				class: [
+					'ck',
+					'ck-list-item-button__check-holder',
+					bind.to( 'isOn', isOn => isOn ? 'ck-on' : 'ck-off' )
+				]
+			}
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		if ( this.isOn ) {
+			this.children.add( this._checkIconView, 0 );
+		}
+
+		this._watchCheckIconMount();
+	}
+
+	/**
+	 * Renders the check icon if the button is toggleable.
+	 */
+	private _watchCheckIconMount() {
+		this.on<ObservableChangeEvent<boolean>>( 'change:isOn', ( evt, propertyName, isOn ) => {
+			const { children, _checkIconView } = this;
+
+			if ( isOn && !children.has( _checkIconView ) ) {
+				children.add( _checkIconView );
+			} else if ( !isOn && children.has( _checkIconView ) ) {
+				children.remove( _checkIconView );
+			}
+		} );
```

Comment:
@scofalik Two things:

1. While it's a simpler approach, I'm not sure if it's the fastest one. Although the check is not rendered (due to `display: none`) it's still attached to the DOM. In smaller lists it makes no difference, but in the bigger lists it may. I assume so because SVG is not a trivial component to render, it's quite complicated, and our UI is not so performant in terms of rendering views.
2. Not every `ListItemButtonView` is checkbox. I don't think we should render unnecessary DOM elements in elements that will never become checkboxes, even if it's simpler, because it can be misleading from an inspector perspective.

Reply:
Sure, that was just my thought, I thought that maybe we do such things somewhere anyway, so I thought to ask about it. I am completely fine with leaving it as it is, or maybe @oleq knows some cleaner way to manage it.

Reply:
Okay, let's leave it as it is, with maybe one nitpick -- I think it is the same situation as with the other place, where `children.has` should be determined by previous `isOn` state. So maybe the `if` can be simplified.

Reply:
Ok, so I'm keeping it as it is